### PR TITLE
WelcomeViewController inheritance

### PR DESCRIPTION
**Pull request by synchuk2010**
### Jira references
- Ticket (issue) — https://essentiatechnologies.atlassian.net/browse/MB-365
### Description
- Move WelcomeViewController inheritance to BaseTableAdapterViewController
- Add TableComponents

### CODE REVIEW

File: Essentia/UserCases/Login/Controllers/WelcomeViewController.swift [hunk 1]
```diff
@@ -34,19 +27,27 @@ class WelcomeViewController: BaseViewController, ImportAccountDelegate, SelectAc
...
-        if (inject() as ViewUserStorageServiceInterface).users.isEmpty {
-            enterButton.setTitle(LS("Welcome.Start"), for: .normal)
-        } else {
-            enterButton.setTitle(LS("Welcome.Enter"), for: .normal)
-        }
-        
+    // MARK: - State
+    override var state: [TableComponent] {
+        return [
+            .empty(height: 30, background: colorProvider.settingsCellsBackround),
```
Comment (impul) [inline Essentia/UserCases/Login/Controllers/WelcomeViewController.swift]: Height of empty space is 40

File: Essentia/UserCases/Login/Controllers/WelcomeViewController.swift [hunk 2]
```diff
@@ -34,19 +27,27 @@ class WelcomeViewController: BaseViewController, ImportAccountDelegate, SelectAc
...
+                                 borderColor: colorProvider.borderedButtonBorderColor.cgColor, borderWidth: 2),
+            .empty(height: 88, background: colorProvider.settingsCellsBackround),
+            .titleWithFontAligment(font: AppFont.regular.withSize(36), title: LS("Welcome.Title1"), aligment: .left, color: colorProvider.appTitleColor),
+            .titleWithFontAligment(font: AppFont.bold.withSize(36), title: LS("Welcome.Title2"), aligment: .left, color: colorProvider.appTitleColor),
+            .calculatbleSpace(background: .clear),
+            .centeredButton(title: (inject() as ViewUserStorageServiceInterface).users.isEmpty ? LS("Welcome.Start") : LS("Welcome.Enter"),
+                            isEnable: true,
+                            action: enterAction,
+                            background: colorProvider.settingsCellsBackround)]
+            + termsState
```
Comment (impul) [inline Essentia/UserCases/Login/Controllers/WelcomeViewController.swift]: Do not separate state if is not necessarily. Put `termsState` to `state`. Attributed text put to private variable.

File: Modules/EssUI/Sources/TableAdapterCore/TableAdapter.swift [hunk 3]
```diff
@@ -273,6 +274,28 @@ public class TableAdapter: NSObject, UITableViewDataSource, UITableViewDelegate
...
+            let cell: TableComponentBorderedButton = tableView.dequeueReusableCell(for: indexPath)
+            cell.titleButton.setTitle(title, for: .normal)
+            cell.drawBorder(color: borderColor, width: borderWidth)
+            cell.action = action
+            return cell
+        case .rightBorderedButton(let title, let action, let borderColor, let borderWidth):
+            let cell: TableComponentBorderedButton = tableView.dequeueReusableCell(for: indexPath)
+            cell.titleButton.setTitle(title, for: .normal)
+            cell.drawBorder(color: borderColor, width: borderWidth)
+            let inset = tableView.frame.width - 115
```
Comment (impul) [inline Modules/EssUI/Sources/TableAdapterCore/TableAdapter.swift]: Make constraints in xib

File: Modules/EssUIGallery/EssUIGallery.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -257,36 +257,13 @@
 			inputFileListPaths = (
 				"${PODS_ROOT}/Target Support Files/Pods-EssUIGallery/Pods-EssUIGallery-frameworks-${CONFIGURATION}-input-files.xcfilelist",
 			);
-			inputPaths = (
```
Comment (impul) [inline Modules/EssUIGallery/EssUIGallery.xcodeproj/project.pbxproj]: The same, reinstall pods

File: Essentia/UserCases/Login/Controllers/WelcomeViewController.swift [hunk 5]
```diff
@@ -61,36 +62,29 @@ class WelcomeViewController: BaseViewController, ImportAccountDelegate, SelectAc
...
-        
-        // MARK: - Colors
-        title1Label.textColor = colorProvider.appTitleColor
-        title2Label.textColor = colorProvider.appTitleColor
-        termsButton.titleLabel?.textColor = colorProvider.appLinkTextColor
         
-        // MARK: - Font
-        title1Label.font = AppFont.regular.withSize(36)
-        title2Label.font = AppFont.bold.withSize(36)
+        return [.attributedCenteredButton(attributedTitle: termsAttributedText, action: termsAction, textColor: colorProvider.appLinkTextColor, background: .clear),
```
Comment (impul) [inline Essentia/UserCases/Login/Controllers/WelcomeViewController.swift]: If all parameters not fit in one line then paste every parameter in signature from new line

Comment (impul) [reply Essentia/UserCases/Login/Controllers/WelcomeViewController.swift]: Same on all lines > 120

File: Essentia/UserCases/Login/Controllers/WelcomeViewController.swift (hunk 1 above)
Comment (synchuk2010) [reply Essentia/UserCases/Login/Controllers/WelcomeViewController.swift]: fixed, ef08e75

File: Essentia/UserCases/Login/Controllers/WelcomeViewController.swift (hunk 2 above)
Comment (synchuk2010) [reply Essentia/UserCases/Login/Controllers/WelcomeViewController.swift]: fixed, ef08e75

File: Modules/EssUI/Sources/TableAdapterCore/TableAdapter.swift (hunk 3 above)
Comment (synchuk2010) [reply Modules/EssUI/Sources/TableAdapterCore/TableAdapter.swift]: fixed, ef08e75

File: Essentia/UserCases/Login/Controllers/WelcomeViewController.swift (hunk 5 above)
Comment (synchuk2010) [reply Essentia/UserCases/Login/Controllers/WelcomeViewController.swift]: fixed, ef08e75

File: Modules/EssUIGallery/EssUIGallery.xcodeproj/project.pbxproj (hunk 4 above)
Comment (synchuk2010) [reply Modules/EssUIGallery/EssUIGallery.xcodeproj/project.pbxproj]: see, 28d559a

File: Essentia.xcodeproj/project.pbxproj [hunk 6]
```diff
@@ -680,42 +668,13 @@
 			inputFileListPaths = (
 				"${PODS_ROOT}/Target Support Files/Pods-Essentia/Pods-Essentia-frameworks-${CONFIGURATION}-input-files.xcfilelist",
 			);
-			inputPaths = (
```
[the remark this answers is not in the document]
Comment (synchuk2010) [reply Essentia.xcodeproj/project.pbxproj]: see, 28d559a

File: Essentia/UserCases/Login/Controllers/WelcomeViewController.swift [hunk 7]
```diff
@@ -34,19 +27,39 @@ class WelcomeViewController: BaseViewController, ImportAccountDelegate, SelectAc
     
     override func viewDidLayoutSubviews() {
```
Comment (impul) [inline Essentia/UserCases/Login/Controllers/WelcomeViewController.swift]: Make table reload in `viewDidLoad`

File: Essentia/UserCases/Login/Controllers/WelcomeViewController.swift [hunk 8]
```diff
@@ -14,14 +14,7 @@ import EssDI
...
-    
-    // MARK: - IBOutlet
-    @IBOutlet weak var restoreButton: UIButton!
-    @IBOutlet weak var title1Label: UILabel!
-    @IBOutlet weak var title2Label: UILabel!
-    @IBOutlet weak var enterButton: CenteredButton!
-    @IBOutlet weak var termsButton: UIButton!
+class WelcomeViewController: BaseTableAdapterController, ImportAccountDelegate, SelectAccountDelegate {
     
     // MARK: - Dependences
```
Comment (impul) [inline Essentia/UserCases/Login/Controllers/WelcomeViewController.swift]: Make all injection of dependencies  in this file under Dependencies mark

File: Essentia/UserCases/Login/Controllers/WelcomeViewController.swift (hunk 7 above)
Comment (synchuk2010) [reply Essentia/UserCases/Login/Controllers/WelcomeViewController.swift]: done, f7d5855

File: Essentia/UserCases/Login/Controllers/WelcomeViewController.swift (hunk 8 above)
Comment (synchuk2010) [reply Essentia/UserCases/Login/Controllers/WelcomeViewController.swift]: done, f7d5855

**Pull request merged**